python/tests/types/test_pandas.py

```diff
@@ -0,0 +1,88 @@
+#  Copyright 2022 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+from rikai.types import Image, Box2d
+from rikai.types.pandas import ImageDtype
+
+
+def test_image(tmp_path):
+    # Convert string to Image
+    s = pd.Series(np.array(['s3://bucket/path/to/image.jpg']),
+                  dtype=ImageDtype())
+    assert s.dtype.name == 'image'
+    assert isinstance(s[0], Image)
+
+    df = pd.DataFrame({'c': s})
+    assert len(df) == len(s)
+    assert df.c.dtype == 'image'
+
+    # Check roundtrip pandas<>arrow
+    table = pa.Table.from_pandas(df)
+    assert len(table) == len(s)
+    arrow_type = table.field('c').type
+    assert arrow_type.extension_name == 'rikai.image'
+    assert arrow_type.storage_type == pa.struct([
+        pa.field('uri', pa.string()),
+        pa.field('data', pa.binary())
+    ])
+    assert arrow_type.to_pandas_dtype().name == 'image'
+
+    df_rt = table.to_pandas()
+    assert len(df_rt) == len(s)
+    assert df_rt.c.dtype == 'image'
+
+    # Check roundtrip pandas<>parquet
+    pq.write_to_dataset(table, str(tmp_path))
+    df_from_parquet = pd.read_parquet(str(tmp_path))
+    assert len(df_from_parquet) == len(s)
+    assert df_from_parquet.c.dtype == 'image'
+    assert isinstance(df_from_parquet.c[0], Image)
+
+
+def test_nested(tmp_path):
```

Comment:
Here’s the test case. I need to dig a little more into arrow internals to see if I can get native support for this kind of nesting to work properly.